### DISCUSSION TITLE
How to get debug information from running unit?

### DISCUSSION THREAD

**Discussion by hekopath** [cyberman54/ESP32-Paxcounter#721] [Q&A]
Hello,

i have two paxcounter units based on ttgo t-beam v1.1 with diffrent settings in use.
One of these devices stopped transmitting messages during battery operation.
I assume an issue with deepsleep and active ADR as this is the main diffrence in the used configuration.
When i establish a serial connection over usb the device will reset.
Is there a best pratice how to determine the status of this unit and gather debug information without this forced reset?

**Comment by cyberman54** [comment]
Difficult, since there is unfortunately no logging mechanism on the board implemented yet, e.g. writing a log file in NVRAM.
Only option you have is to query 0x80 and 0x81 by remote command. The update time value shows, if the board had an unsolicited reset. 

This of course won't work, if the LORA stack hangs. But a failure by LORA stack is signalled by red LED on T-Beam continously flashing.

If you have a display to the T-Beam connected, you can press the middle button to flip display pages to LORA parameters overview. Here you see (if the device is joined) the up- und down frame counters, to compare with counters of LORA network servers.

Note: Deep sleep is a new feature, implemented a few weeks ago. There may be issues, i.e. in conjunction with the LORA stack, since MCCI LMIC not yet supports sleep.

**Comment by cyberman54** [comment]
You also could use a debug probe connected to the board, if you can reproduce the issue. See https://docs.platformio.org/en/latest/plus/debugging.html#tools-debug-probes , i suggest the FT2232H USB Minimodule.

**Reply by hekopath** [reply]
Thank you for your hint.
I will try to build something similar based on a ESP8266 as wireless serialport.
Some of the mentioned debug probes must be here in my lab, somewhere...

**Reply by cyberman54** [reply]
That sounds interesting! Please let us know here, if there is progress. Thanks.
